cosmwasm/contracts/token-bridge/Cargo.toml

```diff
@@ -10,6 +10,9 @@ crate-type = ["cdylib", "rlib"]
 
 [features]
 backtraces = ["cosmwasm-std/backtraces"]
+# This feature flag does a 'shutdown' mode compilation, which disables most
+# functionality except for governance
+shutdown = []
```

Comment:
Cargo features are supposed to be [additive](https://doc.rust-lang.org/cargo/reference/features.html?highlight=additive#feature-unification):

> When a dependency is used by multiple packages, Cargo will use the union of all features enabled on that dependency when building it.
> ...
> A consequence of this is that features should be additive. That is, enabling a feature should not disable functionality, and it should usually be safe to enable any combination of features. A feature should not introduce a [SemVer-incompatible change](https://doc.rust-lang.org/cargo/reference/features.html?highlight=additive#semver-compatibility).

Yes, this means all the crypto crates using features for mutually exclusive things are _wrong_.
 
I think the usual convention for something like this would be to define a "full" feature that's enabled by default:

```
[features]
full = []
default = ["full"]
```

and then the shutdown contract would depend on this crate with default features disabled:

```
[dependencies]
token-bridge-terra-2 = { path = "../token-bridge", default-features = false }
```

Reply:
thanks, I wasn't aware of this, but makes perfect sense